File: fsharp/src/Library/PromptSpec.fs
namespace CallCodingClis

module PromptSpec =

    let buildPromptSpec (prompt: string) : Result<CommandSpec, string> =
        let trimmed = prompt.Trim()
        if System.String.IsNullOrEmpty trimmed then
            Error "prompt must not be empty"
        else
            Ok { Argv = ["opencode"; "run"; trimmed]
                 StdinText = None
                 Cwd = None
                 Env = Map.empty }

File: haskell/src/CallCodingClis/PromptSpec.hs
module CallCodingClis.PromptSpec
  ( buildPromptSpec
  ) where

import CallCodingClis.Types
import Data.Char (isSpace)
import Data.List (dropWhileEnd)

strip :: String -> String
strip = dropWhileEnd isSpace . dropWhile isSpace

buildPromptSpec :: String -> Either String CommandSpec
buildPromptSpec prompt =
  let trimmed = strip prompt
  in if null trimmed
     then Left "prompt must not be empty"
     else Right $ CommandSpec ["opencode", "run", trimmed] Nothing Nothing []
